src/components/Transactions/History/ExportTransactionsHistory.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022-2023 @Polkasafe/polkaSafe-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import dayjs from 'dayjs';
+import React from 'react';
+import CancelBtn from 'src/components/Settings/CancelBtn';
+import AddBtn from 'src/components/Settings/ModalBtn';
+import { EExportType } from 'src/Screens/Transactions';
+
+interface IHistoryTxns {
+	date: Date,
+	amount: string,
+	callhash: string,
+	from: string,
+	token: string,
+	network: string
+}
+
+interface IExportTransactionsHistory {
+	exportType: EExportType,
+	historyTxns?: IHistoryTxns[],
+	onCancel: () => void
+}
+
+const ExportTransactionsHistory = ({ historyTxns, exportType, onCancel }: IExportTransactionsHistory) => {
+	const downloadFile = ({ data, fileName, fileType }: { data: any, fileName: string, fileType: string }) => {
+		// Create a blob with the data we want to download as a file
+		const blob = new Blob([data], { type: fileType });
+		// Create an anchor element and dispatch a click event on it
+		// to trigger a download
+		const a = document.createElement('a');
+		a.download = fileName;
+		a.href = window.URL.createObjectURL(blob);
+		const clickEvt = new MouseEvent('click', {
+			bubbles: true,
+			cancelable: true,
+			view: window
+		});
+		a.dispatchEvent(clickEvt);
+		a.remove();
+	};
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	const exportToCsv = () => {
+		if(!historyTxns) return;
+		// Headers for each column
+		const quickbooksHeaders = ['Date,Type,Doc Number,P0 Number,Name,Source,Account,Detail Account,Job,Item,Class,Cleared,Amount,Memo'];
+		const xeroHeaders = ['Date,Amount,Payee,Description,Reference,Transaction Type,Account Code,Department,Location'];
+
+		// Convert users data to a csv
+		const quickbooksTransactionsCsv = historyTxns.reduce((txns, a, i) => {
+			const { date, callhash, token, from, amount } = a;
+			const formattedDate = dayjs(date).format('L');
+			txns.push([formattedDate, 'General Journal', i+1, i+1, callhash, from, '', '', '', '', token, '', amount, ''].join(','));
```

Comment:
extract `'General Journal'`, `quickbooksHeaders`, `xeroHeaders` to a constants file